napari/layers/points/tests/test_points.py

```diff
@@ -328,55 +331,71 @@ def test_edge_color():
     np.random.seed(0)
     data = 20 * np.random.random(shape)
     layer = Points(data)
-    assert layer.edge_color == 'black'
+    ca_black = ColorArray(['black'] * shape[0])
+    assert layer.edge_color == ca_black[0]
     assert len(layer.edge_colors) == shape[0]
-    assert np.all([col == 'black' for col in layer.edge_colors])
+    np.testing.assert_allclose(ca_black.rgba, layer.edge_colors.rgba)
 
     # With no data selected chaning edge color has no effect
     layer.edge_color = 'blue'
-    assert layer.edge_color == 'blue'
-    assert np.all([col == 'black' for col in layer.edge_colors])
+    assert layer.edge_color == ColorArray('blue')
+    np.testing.assert_allclose(ca_black.rgba, layer.edge_colors.rgba)
 
     # Select data and change edge color of selection
     layer.selected_data = [0, 1]
-    assert layer.edge_color == 'black'
-    layer.edge_color = 'green'
-    assert np.all([col == 'green' for col in layer.edge_colors[:2]])
-    assert np.all([col == 'black' for col in layer.edge_colors[2:]])
+    assert layer.edge_color == ca_black[0]
+    layer.edge_color = ColorArray('green')
+    ca_green = ColorArray(['green'] * len(layer.selected_data))
+    np.testing.assert_allclose(ca_green.rgba, layer.edge_colors[:2].rgba)
+    np.testing.assert_allclose(ca_black[2:].rgba, layer.edge_colors[2:].rgba)
 
     # Add new point and test its color
     coord = [18, 18]
     layer.selected_data = []
     layer.edge_color = 'blue'
     layer.add(coord)
+    ca_black.extend('blue')
     assert len(layer.edge_colors) == shape[0] + 1
-    assert np.all([col == 'green' for col in layer.edge_colors[:2]])
-    assert np.all([col == 'black' for col in layer.edge_colors[2:10]])
-    assert np.all(layer.edge_colors[10] == 'blue')
+    np.testing.assert_allclose(ca_green.rgba, layer.edge_colors[:2].rgba)
+    np.testing.assert_allclose(ca_black[2:].rgba, layer.edge_colors[2:].rgba)
+    np.testing.assert_allclose(
+        ColorArray("blue").rgba, layer.edge_colors[10].rgba
+    )
 
     # Instantiate with custom edge color
     layer = Points(data, edge_color='red')
-    assert layer.edge_color == 'red'
+    np.testing.assert_allclose(layer.edge_color.rgba, ColorArray('red').rgba)
 
     # Instantiate with custom edge color list
     col_list = ['red', 'green'] * 5
     layer = Points(data, edge_color=col_list)
-    assert layer.edge_color == 'black'
-    assert layer.edge_colors == col_list
+    # Modified the behavior of the following assertion since I don't understand
+    # why should we assert that edge_color is black. If it's supposed to
+    # save the last modification of colors, then it should either be the given
+    # col_list or simply empty.
+    # assert layer.edge_color == ca_black[0]
```

Comment:
What do you think regarding this comment?

Reply:
yeah, this is a bit strange default behavior, and made even more confusing by the currently non-ideal names. I think we should change the code so that `edge_color` (really `current_edge_color`) gets set to the color of the lasted added edge in cases like this. We should probably change this convention everywhere in the codebase, but for now just changing it on points for edge and face color is fine